### PR TITLE
chore(all): replace chrono with time

### LINKED ISSUE #2172: Date format is wrong
Our error message is wrong and we accept a format we shouldn’t accept.
The error message in question is:
```
expiresAt field value `xxxxxxxxx` is invalid. It should be in ISO-8601 format to represents a date or datetime in the future or specified as a null value. e.g. 'YYYY-MM-DD' or 'YYYY-MM-DDTHH:MM:SS'.
```

My problem is that...
```
YYYY-MM-DD
YYYY-MM-DDTHH:MM:SS
```

are not valid in ISO-8601. As you can see [here](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) the RFC clearly state that a date time is composed of:
```
   date-time       = full-date "T" full-time
```
A `full-date` followed by a `T` and the followed by a `full-time`.
So obviously this format `YYYY-MM-DD` is wrong since there is no `T` at the end.

And now we’re going to look into the `full-time` thingy:
```
   full-time       = partial-time time-offset
   time-offset     = "Z" / time-numoffset
   time-numoffset  = ("+" / "-") time-hour ":" time-minute
```

Here as you can see a date either end by `Z` or by something complicated with a `+` or `-`.
Whatever, it never end by nothing like that `YYYY-MM-DDTHH:MM:SS`.

### PR DESCRIPTION
**Pull request by irevoire**
Chrono has been unmaintained for a few month now and there is a CVE on it.

Also I updated all the error messages related to the API key as you can see here: https://github.com/meilisearch/specifications/pull/114

fix #2172

### CODE REVIEW

**Comment by curquiza** [comment]
If you want to make it for v0.26.0, you should merge it into `release-v0.26.0` :)

**Comment by irevoire** [comment]
I think we should keep it on main.
I highly doubt there will be any issue with it but since I had to update Meilisearch, milli and segment and around 500loc we never know 👀

File: meilisearch-auth/src/error.rs [hunk 1]
```diff
@@ -10,13 +10,13 @@ pub type Result<T> = std::result::Result<T, AuthControllerError>;
...
     #[error("`{0}` field is mandatory.")]
     MissingParameter(&'static str),
-    #[error("actions field value `{0}` is invalid. It should be an array of string representing action names.")]
+    #[error("`actions` field value `{0}` is invalid. It should be an array of string representing action names.")]
     InvalidApiKeyActions(Value),
-    #[error("indexes field value `{0}` is invalid. It should be an array of string representing index names.")]
+    #[error("`indexes` field value `{0}` is invalid. It should be an array of string representing index names.")]
     InvalidApiKeyIndexes(Value),
-    #[error("expiresAt field value `{0}` is invalid. It should be in ISO-8601 format to represents a date or datetime in the future or specified as a null value. e.g. 'YYYY-MM-DD' or 'YYYY-MM-DDTHH:MM:SS'.")]
+    #[error("`expiresAt` field value `{0}` is invalid. It should follow the RFC 3339 format to represents a date or datetime in the future or specified as a null value. e.g. 'YYYY-MM-DD' or 'YYYY-MM-DD HH:MM:SS'.")]
```
[the remark this answers is not in the document]
Comment (irevoire) [reply meilisearch-auth/src/error.rs:17]: Here is a little sum up of what is accepted and not between the RFC and the ISO format: https://ijmacd.github.io/rfc3339-iso8601/

> does the RFC 3339 accept the YYYY-MM-DD format?

Yes but as a date, not a date-time.

`YYYY-MM-DD HH:MM:SS` Is not valid without the `Z` at the end though but we still want to accept it.

File: meilisearch-http/tests/auth/api_keys.rs [hunk 2]
```diff
@@ -136,6 +136,7 @@ async fn add_valid_api_key_no_description() {
 
     let (response, code) = server.add_api_key(content).await;
 
+    dbg!(&response);
```
[the remark this answers is not in the document]
Comment (irevoire) [reply meilisearch-http/tests/auth/api_keys.rs:139]: Thanks!

**Comment by curquiza** [comment]
@irevoire can you confirm this PR solves this issue https://github.com/meilisearch/meilisearch/issues/2172?
If yees, can you link the issue in this PR?

**Review by ManyTheFish — APPROVED** [review]

**Comment by irevoire** [comment]
@curquiza yes thank you I forgot :+1: 

bors merge

**Pull request merged**